app/controllers/kba_security_controller.rb

```diff
@@ -0,0 +1,49 @@
+class KbaSecurityController < ApplicationController
+  def show
+    @kba_security_form = kba_security_form
+    redirect_to root_url unless @kba_security_form.user
+  end
+
+  def update
+    result = kba_security_form.submit(params[:kba_security_form])
+    return if handle_errors(result)
+    user = result.extra[:user]
+    if result.success?
+      handle_success(user, kba_security_form)
+    else
+      handle_failure(user, kba_security_form)
+    end
+    redirect_to root_url
+  end
+
+  private
+
+  def handle_errors(result)
+    errors = result.errors
+    error_message = errors[:answer]
+    if error_message
+      flash[:error] = error_message
+      redirect_to change_phone_url(token: params[:kba_security_form][:token])
+      return true
+    end
+    false
+  end
+
+  def handle_success(user, kba_security_form)
+    answer = kba_security_form.answer
+    analytics.track_event(Analytics::RESET_DEVICE_CORRECT_SECURITY_ANSWER)
```

Comment:
Instead of creating separate analytics events for each possible outcome, what do you think of following our convention where we have one analytics event per form, and then we track the various attributes of the event, such as whether or not it was successful?

Something like this:
```ruby
def update
  result = kba_security_form.submit(params[:kba_security_form])
  analytics.track_event(Analytics::RESET_DEVICE, result.to_h)
   ...
end
```
and in the Form Object, you could capture other things we might want to track.

Reply:
My only concern is that we have to monitor RESET_DEVICE_WRONG_SECURITY_ANSWER and RESET_DEVICE_REPORTED_FRAUD individually since these could indicate potential attacks.  Do we normally take action on specific hash/results of an event?  If so, then I'm ok with going that route.

Reply:
Yes, Jordan can create an alert for whatever query we want. This is how we've always done this. For example, we have a metric that looks at failed authentication. On the analytics side, it's just one authentication event with different attributes, such as passed or failed, and within failed it could be due to existing user with invalid credentials, or nonexistent user, or blocked user, etc.

Reply:
Sounds good. I'll make the changes.